apps/meteor/ee/app/livechat-enterprise/server/api/lib/priorities.ts

```diff
@@ -46,6 +39,78 @@ export async function findPriorities({ text, pagination: { offset, count, sort }
 	};
 }
 
-export async function findPriorityById({ priorityId }: FindPrioritiesByIdParams): Promise<FindPrioritiesByIdResult> {
-	return LivechatPriority.findOneById(priorityId);
+export async function updatePriority(
+	_id: string,
+	data: Pick<ILivechatPriority, 'name'> & { reset?: boolean },
+): Promise<ILivechatPriority | null> {
+	const query = {
+		_id,
+	};
+	const update: Pick<UpdateFilter<ILivechatPriority>, '$set' | '$unset'> = {
+		...((data.reset && {
+			$set: { dirty: false },
+			$unset: { name: 1 },
+		}) || {
+			// Trim value before inserting
+			$set: { name: data.name?.trim(), dirty: true },
+		}),
+	};
+
+	if (data.name) {
+		// If we want to enforce translated duplicates we need to change this
+		const priority = await LivechatPriority.findOne({ name: new RegExp(`^${escapeRegExp(data.name.trim())}$`, 'i') });
+		if (priority && priority._id !== _id) {
+			throw new Error('error-duplicate-priority-name');
+		}
+	}
+
+	const createdResult = await LivechatPriority.findOneAndUpdate(query, update, {
```

Comment:
can you create a model method for this action? just like `setSlaForRoom` and `findNextAndLock` .. ideally we should not use collection method outside the model. 

also, bringing the var definitions closer to the function call helps readability (in this case `query` and `update` and defined far away from here, where they're being used)